platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/SolrHttpRequestRetryHandler.java

```diff
@@ -43,14 +43,17 @@ public boolean retryRequest(IOException e, int retryCount, HttpContext httpConte
     LOGGER.debug("Connection failed", e);
     try {
       long waitTime = (long) Math.pow(2, Math.min(retryCount, MAX_RETRY_COUNT)) * 50;
-      LOGGER.debug(
-          "Solr Client: Connection failed, waiting {} before retrying.",
-          DurationFormatUtils.formatDurationWords(waitTime, true, true));
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug(
+            "Solr Client: Connection failed, waiting {} before retrying.",
+            DurationFormatUtils.formatDurationWords(waitTime, true, true));
+      }
       synchronized (this) {
         wait(waitTime);
       }
     } catch (InterruptedException ie) {
       LOGGER.debug("Exception while waiting.", ie);
+      Thread.currentThread().interrupt();
```

Comment:
❗️ Dangerous without checking how this method is called as if it is called within a loop, it will continuously get interrupted. We have to figure out how to properly handle the interruptions and it might turn out that simply propagating it is ok.